### ISSUE TITLE
Move more consistency checks to the contract state machine

### ISSUE THREAD

**Issue by paulbrauner-da** [digital-asset/daml#17396]

**Comment by dylant-da** [comment]
This doesn't appear to be done, the linked PR says it's a "warm-up"
@paulbrauner-da to advise further

**Comment by paulbrauner-da** [comment]
Indeed, the real PR is https://github.com/digital-asset/daml/pull/18474 but hasn't been merged (was pre-empted by more urgent things). As part of NUCK @remyhaemmerle-da will refactor the contract state machine, and he confirmed to me yesterday that the new version will check for double-spends. So maybe we can close it or re-assign it to @remyhaemmerle-da.

Remy, what do you think?

**Comment by paulbrauner-da** [comment]
Will be addressed as part of NUCK design.
